core/watcher/eventqueue/notify.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventqueue
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/errors"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/changestream"
+)
+
+// NotifyWatcher watches for changes to a database table.
+type NotifyWatcher struct {
+	*BaseWatcher
+
+	out       chan struct{}
+	tableName string
+	selectAll string
+}
+
+// NewNotifyWatcher returns a new watcher that receives changes ...
+func NewNotifyWatcher(base *BaseWatcher, tableName, scope string) *NotifyWatcher {
+	w := &NotifyWatcher{
+		BaseWatcher: base,
+		out:         make(chan struct{}),
+		tableName:   tableName,
+		selectAll:   fmt.Sprintf("SELECT %s FROM %s", scope, tableName),
+	}
+
+	w.tomb.Go(w.loop)
```

Comment:
this is the BaseWatcher's tomb, not sure on how the notifywatcher is created or the base watcher is created, but we need to be careful here.